modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/action/ViewFileEntryContentDashboardItemAction.java

```diff
@@ -96,26 +95,33 @@ private String _getViewURL(Locale locale, ThemeDisplay themeDisplay) {
 
 			clonedThemeDisplay.setScopeGroupId(_fileEntry.getGroupId());
 
-			return Optional.ofNullable(
+			String friendlyURL =
 				_assetDisplayPageFriendlyURLProvider.getFriendlyURL(
 					FileEntry.class.getName(), _fileEntry.getFileEntryId(),
-					locale, clonedThemeDisplay)
-			).map(
-				url -> {
-					String backURL = ParamUtil.getString(
-						_httpServletRequest, "backURL");
-
-					if (Validator.isNotNull(backURL)) {
-						return HttpComponentsUtil.setParameter(
-							url, "p_l_back_url", backURL);
-					}
-
-					return HttpComponentsUtil.setParameter(
-						url, "p_l_back_url", themeDisplay.getURLCurrent());
-				}
-			).orElse(
-				StringPool.BLANK
-			);
+					locale, clonedThemeDisplay);
+
+			if (friendlyURL == null) {
+				return StringPool.BLANK;
+			}
+
+			String url = null;
```

Comment:
@slnn This variable is not needed. If you just keep re-assigning `friendlyURL` you will be able to simplify logic to
```
if (xxx) {
    return xxxxx1;
}

return xxxxx2;
```

Also, if friendlyURL is not null, the later operations will never return `null`, so the null check on line 120 is redundant.